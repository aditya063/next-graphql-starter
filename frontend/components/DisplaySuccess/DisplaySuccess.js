import PropTypes from 'prop-types';

import * as sc from './DisplaySuccess.style';

const DisplaySuccess = React.memo(props => (
  <sc.DisplaySuccess>
    <li>{props.message}</li>
  </sc.DisplaySuccess>
));

DisplaySuccess.propTypes = {
  message: PropTypes.string.isRequired
};

export default DisplaySuccess;
