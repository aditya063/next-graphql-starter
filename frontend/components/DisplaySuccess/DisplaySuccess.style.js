import styled from 'styled-components';

export const DisplaySuccess = styled.ul`
  border-left: 5px solid ${props => props.theme.color.green};
  list-style: none;
  padding-left: 1rem;
`;
