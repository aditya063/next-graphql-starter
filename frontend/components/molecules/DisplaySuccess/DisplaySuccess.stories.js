import DisplaySuccess from './DisplaySuccess';

export default { title: 'Molecules', component: DisplaySuccess };

const data = {
  message: 'this is a message',
  type: 'success'
};

export const displaySuccess = () => <DisplaySuccess {...data} />;
