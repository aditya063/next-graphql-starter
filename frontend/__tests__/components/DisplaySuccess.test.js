import { prettyDOM, render, fireEvent, wait } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import DisplaySuccess from '../../components/DisplaySuccess/DisplaySuccess';
import theme from '../../public/styles/theme.style';

describe('DisplaySuccess', () => {
  it('renders message', () => {
    const successText = 'mock success message';
    const mockProps = { message: successText };

    const utils = render(
      <ThemeProvider theme={theme}>
        <DisplaySuccess {...mockProps} />
      </ThemeProvider>
    );

    const container = utils.container.firstChild;

    expect(container).toMatchSnapshot();
    expect(container).toHaveTextContent(successText);
  });
});
