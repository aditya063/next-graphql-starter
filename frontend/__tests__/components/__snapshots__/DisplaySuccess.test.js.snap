// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplaySuccess renders message 1`] = `
.c0 {
  border-left: 5px solid limegreen;
  list-style: none;
  padding-left: 1rem;
}

<ul
  class="c0"
>
  <li>
    mock success message
  </li>
</ul>
`;
